docs/SDK_API_Definition.md

```diff
@@ -0,0 +1,423 @@
+* `MESH_TIMEOUT_INFINITE` – No timeout, blocks until success or an error.
+* `MESH_TIMEOUT_ZERO` – Polling mode, returns immediately.
+
+### Returns
+0 if successful. Otherwise, returns an [Error code](#return-error-codes).
+
+
+## MeshBuffer
+
+```c
+typedef struct{
+    MeshConnection * const conn;
+    void * const payload_ptr;
+    const size_t payload_len;
+    void * const metadata_ptr;
+    const size_t metadata_len;
+} MeshBuffer;
+```
+### Fields
+* `conn` – Pointer to a parent connection structure.
+* `payload_ptr` – Pointer to shared memory area storing payload data.
+* `payload_len` – Actual length of payload data in the buffer.
+* `metadata_ptr` – Pointer to shared memory area storing metadata.
+* `metadata_len` – Actual length of metadata in the buffer.
+
+
+## mesh_put_buffer()
+```c
+int mesh_put_buffer(MeshBuffer **buf)
+```
+Puts the buffer to the media connection.
+
+### Parameters
+* `[IN/OUT]` `buf` – Address of a pointer to a mesh buffer structure.
+
+### Returns
+0 if successful. Otherwise, returns an [Error code](#return-error-codes).
+
+
+## mesh_put_buffer_timeout()
+```c
+int mesh_put_buffer_timeout(MeshBuffer **buf,
+                            int timeout_us)
+```
+Puts the buffer to the media connection with a timeout.
+
+### Parameters
+* `[IN/OUT]` `buf` – Address of a pointer to a mesh buffer structure.
+* `[IN]` `timeout_us` – Timeout interval in microseconds.
+
+### Returns
+0 if successful. Otherwise, returns an [Error code](#return-error-codes).
+
+
+## mesh_buffer_set_payload_len()
+```c
+int mesh_buffer_set_payload_len(MeshBuffer *buf,
+                                size_t len)
+```
+Sets the payload length in the buffer provided by the media connection.
+
+### Parameters
+* `[IN]` `buf` – Pointer to a mesh buffer structure.
+* `[IN]` `len` – Payload length in bytes.
+
+### Returns
+0 if successful. Otherwise, returns an [Error code](#return-error-codes).
+
+
+## mesh_buffer_set_metadata_len()
+```c
+int mesh_buffer_set_metadata_len(MeshBuffer *buf,
+                                 size_t len)
+```
+Sets the metadata length in the buffer provided by the media connection.
+
+### Parameters
+* `[IN]` `buf` – Pointer to a mesh buffer structure.
+* `[IN]` `len` – Metadata length in bytes.
+
+### Returns
+0 if successful. Otherwise, returns an [Error code](#return-error-codes).
+
+
+## mesh_err2str()
+```c
+const char * mesh_err2str(int err)
+```
+Gets a text description of the error code.
+
+### Parameters
+* `[IN]` `err` – Error code returned from any Mesh DP API call.
+
+### Returns
+NULL-terminated string describing the error code.
+
+
+## Return error codes
+
+API SDK functions may return the following error codes.
+
+NOTE: The codes a negative integer values.
+
+| Error code macro name            | Meaning                        |
+|----------------------------------|--------------------------------|
+| `-MESH_ERR_BAD_CLIENT_PTR`       | Bad client pointer             |
```

Comment:
I think a brief description/possible solutions should be put for some errors here.
f.e. -MESH_ERR_FOUND_ALLOCATED, does not specify  what resources are causing a problem and how to deal with this error.